includes/class-amp-story-post-type.php

```diff
@@ -1553,4 +1637,204 @@ public static function add_new_max_image_size( $image_sizes ) {
+	}
+
+	/**
+	 * Get the args used during a story export.
+	 *
+	 * @param string $slug The slug used to build the new `canonical_url`.
+	 *
+	 * @return array
+	 */
+	public static function get_export_args( $slug = '' ) {
+		$base_url = untrailingslashit( AMP_Options_Manager::get_option( 'story_export_base_url' ) );
+
+		return [
+			'base_url'      => esc_url( $base_url ),
+			'canonical_url' => ( $base_url && $slug ) ? esc_url( trailingslashit( $base_url ) . $slug ) : false,
+		];
+	}
+
+	/**
+	 * Ajax handler to export the story ZIP archive.
+	 *
+	 * This method returns an error as JSON and the binary data on success.
+	 */
+	public static function handle_export() {
+		check_ajax_referer( self::AMP_STORIES_AJAX_ACTION, 'nonce' );
+
+		// The user must have the correct permissions.
+		if ( ! current_user_can( 'edit_files' ) ) {
+			wp_send_json_error(
+				[
+					'errorMessage' => esc_html__( 'You do not have the required permissions to export AMP stories.', 'amp' ),
+				],
+				403
+			);
+		}
+
+		// We need the ZipArchive class to make this work.
+		if ( ! class_exists( 'ZipArchive', false ) ) {
+			wp_send_json_error(
+				[
+					'errorMessage' => esc_html__( 'The ZipArchive class is required to export AMP stories.', 'amp' ),
+				],
+				403
+			);
+		}
+
+		// Bail if the user has not saved the story yet.
+		if ( 'auto-draft' === get_post_status( wp_unslash( $_POST['post_ID'] ) ) ) {
+			wp_send_json_error(
+				[
+					'errorMessage' => esc_html__( 'Save the AMP story before exporting.', 'amp' ),
+				],
+				403
+			);
+		}
+
+		// Generate and export the archive.
+		$export = self::generate_export( wp_unslash( $_POST['post_ID'] ) );
+
+		// Export failed.
+		if ( is_wp_error( $export ) ) {
+			wp_send_json_error(
+				[
+					'errorMessage' => $export->get_error_message(),
+				],
+				403
+			);
+		}
+
+		// Failed to export for an unknown reason not related to generating the archive.
+		wp_send_json_error(
+			[
+				'errorMessage' => esc_html__( 'Could not generate the AMP story archive.', 'amp' ),
+			],
+			403
+		);
+	}
+
+	/**
+	 * Generates a Zip archive from the AMP Story.
+	 *
+	 * @param int $post_id The post ID of the AMP Story.
+	 * @return WP_Error
+	 */
+	private static function generate_export( $post_id ) {
+		$post = get_post( $post_id );
+
+		if ( ! $post ) {
+			return new WP_Error( 'amp_story_export_invalid_post', esc_html__( 'The AMP Story does not exist.', 'amp' ) );
+		}
+
+		$slug = sanitize_title( $post->post_title, $post->ID );
+		$file = wp_tempnam( $slug );
+
+		$zip = new ZipArchive();
+		$res = $zip->open( $file, ZipArchive::CREATE | ZipArchive::OVERWRITE );
+
+		// Generate ZIP and send headers.
+		if ( true === $res ) {
+
+			// Create the zip directory.
+			$zip->addEmptyDir( $slug );
+
+			// Add README.txt file.
+			$zip->addFromString( $slug . '/README.txt', 'temporary content...' );
```

Comment:
**To do:** We need to define the content here.

Also, it might be easier to have this as an actual file in the plugin that we can just include here. Makes changing the text easier.

Reply:
I think this is probably the last piece to get finished. But I agree, a file would make sense.